src/sensai/featuregen.py

```diff
@@ -24,28 +24,45 @@ class FeatureGenerator(ABC):
     from an input DataFrame
     """
     def __init__(self, categoricalFeatureNames: Sequence[str] = (),
-            normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (), addCategoricalDefaultRules=True):
+                 normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (),
+                 generatedColumnsRuleTemplate: data_transformation.DFTNormalisation.RuleTemplate = None, addCategoricalDefaultRules=True):
         """
         :param categoricalFeatureNames: if provided, will ensure that the respective columns in the generated data frames will
             have dtype 'category'.
             Furthermore, presence of meta-information can later be leveraged for further transformations, e.g. one-hot encoding.
         :param normalisationRules: Rules to be used by DFTNormalisation (e.g. for constructing an input transformer for a model).
             These rules are only relevant if a DFTNormalisation object consuming them is instantiated and used
             within a data processing pipeline. They do not affect feature generation.
+        :param generatedColumnsRuleTemplate: If given, a normalisation rule based on this template will be created for all generated columns.
+            As with other rules, this is only relevant if DFTNormalization is used after feature generation.
+            The rule template should only be provided if no categoricalFeatureNames and normalisationRules have been passed
         :param addCategoricalDefaultRules:
             If True, normalisation rules for categorical features (which are unsupported by normalisation) and their corresponding one-hot
             encoded features (with "_<index>" appended) will be added.
         """
+        self.generatedColumnNames = None
+        self._generatedColumnsRule = None
+        self.generatedColumnsRuleTemplate = generatedColumnsRuleTemplate
         self._categoricalFeatureNames = categoricalFeatureNames
         normalisationRules = list(normalisationRules)
         if addCategoricalDefaultRules and len(categoricalFeatureNames) > 0:
             normalisationRules.append(data_transformation.DFTNormalisation.Rule(r"(%s)" % "|".join(categoricalFeatureNames), unsupported=True))
             normalisationRules.append(data_transformation.DFTNormalisation.Rule(r"(%s)_\d+" % "|".join(categoricalFeatureNames), skip=True))
         self._normalisationRules = normalisationRules
 
+        self._inputValidation()
+
+    def _inputValidation(self):
```

Comment:
This should be done directly in the constructor and not based on the members but the actual parameters. Make it the first thing the constructor does.

Reply:
You are not actually checking the second condition you mentioned in the docstring: no categorical features. Then again, we don't necessarily need to. We could make the template apply to all non-categorical features only; that would be the most flexible way of handling things.

Reply:
By checking after all normalisation rules have been set, I also check against rules for categorical features, right? That's why I did it last and based on the members.

Applying the template to non-categorical features only makes the template unusable for declaring normalisation categorical input as unsupported, as would be needed by featuregen from columngen and possibly others. Nevertheless, we can consider applying rule templates only to non-categorical features and handling the categorical ones separately

Reply:
Categorical features are always unsupported; that's what the default rule that is auto-generated states.

Reply:
Anyway, since auto-generation of rules for categorical features always makes sense, the rule template should not apply to them and there isn't really a reason why a rule template should not be usable for the remaining columns in such cases.

Reply:
I adjusted the mechanism of the rule template such that it only applies to non-categorical features